drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/exporters/cde_export.rb

```diff
@@ -59,13 +66,26 @@ def columns
     def cdes
       @cdes ||= Hmis::Hud::CustomDataElement.
         where(data_source: data_source).
+        joins(:data_element_definition).
+        # Exclude some Custom Data Elements that are exported in other files. It would be simpler to
+        # just export them here, but customer prefers to keep dedicated file (which was added first).
+        where.not(data_element_definition: { key: HmisExternalApis::AcHmis::Exporters::CdedExport::EXCLUDED_CUSTOM_DATA_ELEMENT_KEYS }).
         preload(:data_element_definition)
     end
 
     def walk_in_cded
       @walk_in_cded ||= Hmis::Hud::CustomDataElementDefinition.where(data_source: data_source, key: :direct_entry, owner_type: 'Hmis::Hud::Project').first!
     end
 
+    # { source client id => destination warehouse id }
+    # only includes source clients that are the Owner of any CDEs in scope
+    def client_id_to_warehouse_id
+      @client_id_to_warehouse_id ||= begin
+        source_client_ids = cdes.where(data_element_definition: { owner_type: 'Hmis::Hud::Client' }).pluck(:owner_id)
```

Comment:
The id set of all CDEs could be kind of large right? Maybe distinct so there's only one owner (client_id)
```suggestion
        source_client_ids = cdes.where(data_element_definition: { owner_type: 'Hmis::Hud::Client' }).distinct.pluck(:owner_id)
```

Reply:
Thanks, this could definitely be improved! I updated this to just join the client scope to `custom_data_elements` which should have the same effect of limiting scope to only clients that are "owners" of custom data elements. https://github.com/greenriver/hmis-warehouse/pull/5068/commits/ef2cc0dc131777e2d40fb8957c28d8b5142a7fee